lib/ui/platform_dispatcher.dart

```diff
@@ -1269,6 +1280,40 @@ class PlatformDispatcher {
 
   @Native<Handle Function()>(symbol: 'PlatformConfigurationNativeApi::DefaultRouteName')
   external static String _defaultRouteName();
+
+  /// Returns a [TextScaler] that scales font size according to the user's
+  /// platform preference.
+  ///
+  /// Many platforms allow users to scale text globally for better readability.
+  /// Given the font size the app developer specified, this [TextScaler]
+  /// converts it to the preferred font size that accounts for platform-wide
+  /// text scaling.
+  ///
+  /// The scaled value of the same font size input may change if the user changes
+  /// the text scaling preference (in system settings for example). The
+  /// [onTextScaleFactorChanged] callback can be used to monitor such changes.
+  ///
+  /// Instead of directly using this getter, applications should typically use
+  /// `MediaQuery.scaledFontSizeOf` instead to retrive the scaled font size in a
+  /// widget tree, so text resizes properly when the text scaling preference
+  /// changes.
+  ///
+  /// See also:
+  ///
+  ///  * [WidgetsBindingObserver], for a mechanism at the widgets layer to
+  ///    observe when this callback is invoked.
+  TextScaler get platformTextScaler {
```

Comment:
Just to double check: This is on the PlatformDispatcher because it is a global thing for the entire app? This is not going to be a per-view setting?

Reply:
It's been confirmed that this is per Android device not per monitor.  I was also told device manufacturers will not be able to override the text scaling curve: maybe we can avoid using FFI/JNI and just hard-code the formula in dart.  

For that we will need the Android build version and potentially some other information (it sounds like the curve is going to be different on TVs / watches than on phones). I think I'll ask for more information before proceeding since that sounds like a more performant and eaiser to implement solution than this.

Reply:
If I add the `TextScaler` interface to the framework, what would be the easiest way to move it to dart:ui?